### PR TITLE
RD-1706-add-non-recoverable-errors __NODOCS__

### PR DESCRIPTION
**Pull request by ahmadiesa-abu**

### CODE REVIEW

**Review by EarthmanT — APPROVED** [review]

File: fabric_plugin/tests/test_fabric_plugin.py [hunk 1]
```diff
@@ -54,7 +54,8 @@ def _execute(self,
                  process=None,
                  ip=None,
                  custom_input='value',
-                 use_sudo=False):
+                 use_sudo=False,
+                 non_recoverable_errors=[]):
```
Comment (AdarShaked) [inline fabric_plugin/tests/test_fabric_plugin.py:58]: Please do not use immutable types as default values. 
```suggestion
                 non_recoverable_errors=None):
                 non_recoverable_errors = non_recoverable_errors or []
```

**Review by AdarShaked — COMMENTED** [review]
Please add documentation for that feature.
@ahmadiesa-abu

File: fabric_plugin/tests/blueprint/blueprint.yaml [hunk 2]
```diff
@@ -13,6 +13,7 @@ inputs:
     custom_input:
         default: value
     use_sudo:        {}
+    non_recoverable_errors:     {}
```
Comment (AdarShaked) [inline fabric_plugin/tests/blueprint/blueprint.yaml:16]: The name is bit confusing, maybe use :
non_recoverable_error_exit_codes
 what do you think ?
@EarthmanT

**Review by AdarShaked — CHANGES_REQUESTED** [review]
Change the immutable parameter

**Review by AdarShaked — COMMENTED** [review]
.

File: fabric_plugin/tests/blueprint/blueprint.yaml (hunk 2 above)
Comment (EarthmanT) [reply fabric_plugin/tests/blueprint/blueprint.yaml:16]: Sounds good. Shouldn't it be a list?

Comment (AdarShaked) [reply fabric_plugin/tests/blueprint/blueprint.yaml:16]: @ahmadiesa-abu

**Review by EarthmanT — APPROVED** [review]

**Pull request merged**